src/codegen/utilities/interpreter_helpers.py

```diff
@@ -130,7 +132,14 @@ def get_argument_types(functions_metadata):
             if param.is_pointer:
                 argtypes.append(f"ctypes.POINTER({to_param_argtype(param)})")
             else:
-                argtypes.append(to_param_argtype(param))
+                # This is special case. The ctype of these parameters is passed as bool in c call.
+                if (
+                    functions_metadata.function_name in CTYPE_CHANGE_FUNCTIONS
+                    and param.parameter_name in ("read_array", "write_array")
+                ):
+                    argtypes.append("wrapped_ndpointer(dtype=bool, flags=('C'))")
```

Comment:
Read and write need different flags: https://numpy.org/doc/stable/reference/routines.ctypeslib.html#numpy.ctypeslib.ndpointer

https://github.com/ni/nidaqmx-python/blob/0.7.0/generated/nidaqmx/_task_modules/read_functions.py
```
wrapped_ndpointer(dtype=bool, flags=('C', 'W')),
```

https://github.com/ni/nidaqmx-python/blob/0.7.0/generated/nidaqmx/_task_modules/write_functions.py#L229
```
wrapped_ndpointer(dtype=bool, flags=('C')),
```

If you move the special case to scrapigen, I think this code will do the right thing.

Reply:
Moved this logic to Scarpigen. 
I was exploring on these write functions and this PR https://github.com/ni/nidaqmx-python/pull/152 for https://github.com/ni/nidaqmx-python/issues/151 came up, So as expected there, I have made the following changes
1) Differentiated Read and Write functions separately in `functions_addon.py`
2) Passed in Optional parameter `flags="'C','W'"` in `to_param_argtype` and passed `flags` as `"'C'"` for write functions alone
3) Added `is_custom_write_function` method
4) Updated the places accordingly where `CustomCode_Read_Write` is used.